src/components/rainbow-toast/ToastSFSymbolIcon.tsx

```diff
@@ -0,0 +1,71 @@
+import { TOAST_ICON_SIZE } from '@/components/rainbow-toast/constants';
+import { useToastColors } from '@/components/rainbow-toast/useToastColors';
+import { Text } from '@/design-system';
+import React from 'react';
+import { View } from 'react-native';
+
+const sfSymbols = {
+  check: '􀆅',
+  exclamationMark: '􀅎',
+};
+
+export function ToastSFSymbolIcon({
+  name,
+  borderRadius = 100,
+  size = TOAST_ICON_SIZE,
+}: {
+  name: keyof typeof sfSymbols;
+  size?: number;
+  borderRadius?: number;
+}) {
+  const colors = useToastColors();
+  const bg = name === 'check' ? colors.green : colors.red;
+  const content = sfSymbols[name];
+
+  return (
+    <View
+      style={{
+        width: size,
+        height: size,
+        borderRadius: borderRadius,
+        borderWidth: 2,
+        borderColor: bg,
+        shadowColor: bg,
+        shadowRadius: 8,
+        shadowOpacity: 0.4,
+        shadowOffset: { height: 3, width: 0 },
+      }}
+    >
+      <View
+        style={[
+          {
+            flex: 1,
+            backgroundColor: bg,
+            // this gets us the right inner rounding based on the borderWidth above
+            // we use a semi-rounded borderRadius for extral contracts which needs it
+            borderRadius: borderRadius - 4,
+            overflow: 'hidden',
+            opacity: 0.85,
+            alignItems: 'center',
+            justifyContent: 'center',
+            transform: [{ scale: 1.02 }],
+          },
+        ]}
+      >
+        {typeof content === 'string' ? (
+          <Text
+            size="12pt"
+            color={{ custom: colors.white }}
+            weight="bold"
+            // somehow this does make a difference, 0.25 specifically vs 0.5 in centering it right
+            style={{ transform: [{ translateX: 0.25 }] }}
```

Comment:
The reason is because the smallest representable unit on most devices is 1/3 pixel. Typically you want to use Stylesheet.hairlineWidth to represent that unit. Some SF icons are a little hard to center properly and there is a specific <TextIcon> component and size="icon 13px" prop that might help not have to have this transform here, but not sure.